dive/image/docker/image_archive.go

```diff
@@ -76,6 +77,49 @@ func NewImageArchive(tarFile io.ReadCloser) (*ImageArchive, error) {
 
 				// add the layer to the image
 				img.layerMap[tree.Name] = tree
+			} else if strings.HasPrefix(name, "blobs/sha256/") {
+				fileBuffer, err := io.ReadAll(tarReader)
+				if err != nil {
+					return img, err
+				}
+
+				// if the first two bytes of a string are \x1F\x8B then it's tar.gz file
+				if strings.HasPrefix(string(fileBuffer), "\x1F\x8B") {
```

Comment:
unfortunate that your PR isn't getting any attention :-(  but while I'm here, a comment:

What are the memory usage implications of this and the similar check below? Does that create a copy of the []byte content as a string? It could potentially be huge (multiple gigabytes), and we only need to check the first few bytes here...